### ISSUE TITLE
Content type for generated client

### ISSUE THREAD

**Issue by lehejcek** [Carapacik/swagger_parser#45]
Swagger Parser does not seem to respect content type for generated client of the following example. I have to add `contentType: 'application/x-www-form-urlencoded'` and ` data: _data.toJson()` to the generated client or am i missing something? 

```yaml
# https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md
openapi: 3.0.3
info:
  title: Keycloak Token
  version: 1.0.0
servers:
  - url: 'http://localhost:8009/auth'
tags:
  - name: KeycloakToken
paths:
  /realms/{realm}/protocol/openid-connect/token:
    post:
      tags:
        - KeycloakToken
      requestBody:
        required: true
        content:
          # Bug: Content-Type missing after code generation
          # _KeycloakTokenClient
          # contentType: 'application/x-www-form-urlencoded',
          # data: _data.toJson(),
          application/x-www-form-urlencoded:
            schema:
              $ref: '#/components/schemas/KeycloakAuthenticationIdentifier'
      operationId: getKeycloakToken
      parameters:
        - name: realm
          in: path
          required: true
          example: testing
          schema:
            type: string
      responses:
        '200':
          description: OK
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/KeycloakOpenIdentityConnect'
components:
  schemas:
    KeycloakOpenIdentityConnect:
      type: object
      properties:
        access_token:
          type: string
        expires_in:
          type: integer
          example: 300
        refresh_expires_in:
          type: integer
          example: 1800
        refresh_token:
          type: string
        token_type:
          type: string
          example: Bearer
        scope:
          type: string
          example: 'profile email'
      required:
        - access_token
        - expires_in
        - refresh_expires_in
        - refresh_token
        - token_type
        - scope
    KeycloakAuthenticationIdentifier:
      type: object
      properties:
        client_id:
          type: string
          description: The ID of the client.
          example: 'testing-client'
        client_secret:
          type: string
          description: The secret of the client.
          example: '**********'
        username:
          type: string
          description: The username of the user.
          example: 'alice'
        password:
          type: string
          description: The password of the user.
          example: '123456'
        grant_type:
          type: string
          description: The grant type.
          example: 'password'
      required:
        - client_id
        - client_secret
        - username
        - password
        - grant_type
```

**Comment by Carapacik** [comment]
Try to add `@FormUrlEncoded()` to your method. I think this is a bug, since this is also a separate case like `multipart`.
```dart
import 'package:dio/dio.dart';
import 'package:retrofit/retrofit.dart';

import '../shared_models/keycloak_authentication_identifier.dart';
import '../shared_models/keycloak_open_identity_connect.dart';

part 'keycloak_token_client.g.dart';

@RestApi()
abstract class KeycloakTokenClient {
  factory KeycloakTokenClient(Dio dio, {String baseUrl}) = _KeycloakTokenClient;

  @FormUrlEncoded() // <---- add this
  @POST('/realms/{realm}/protocol/openid-connect/token')
  Future<KeycloakOpenIdentityConnect> getKeycloakToken({
    @Path('realm') required String realm,
    @Body() required KeycloakAuthenticationIdentifier body,
  });
}
```

**Comment by lehejcek** [comment]
Currently i see `Package not available (the pubspec for swagger_parser 1.0.2 from git has version 1.0.1).` Already tried `flutter clean`. 

And thanks for instantly providing a fix 💙
